content/en/docs/concepts/security/security-checklist.md

```diff
@@ -0,0 +1,140 @@
+---
+reviewers:
+title: Security Checklist
+description: >
+  Recommended checklist for ensuring security in Kubernetes clusters.
+content_type: concept
+---
+
+<!-- overview -->
+
+This checklist aims at providing a basic list of guidance with links to more
+comprehensive documentation on each topic. It does not claim to be exhaustive
+and is meant to evolve.
+
+The order of topics does not reflect an order of priority.
+
+<!-- body -->
+
+## Authentication & Authorization
+
+See the [Role Based Access Control Good
+Practices](/docs/concepts/security/rbac-good-practices/) for information
+related to authentication and authorization.
+
+<!-- New authentication certificates should be generated on new clusters that avoid -->
+<!-- the use of the `system:masters`. `system:masters` should be used as a -->
+<!-- break-glass mechanism as opposed to an admin user. -->
+
+## Network security
+
+A number of Container Network Interfaces (CNIs) provide functionality to
+restrict network resources pods may communicate with. This is most commonly done
+through [Network Policies](/docs/concepts/services-networking/network-policies/)
+which provide a namespaced resource to define rules.
+
+Not all CNIs provide encryption in transit. If the chosen CNI lacks this
+feature, an alternative solution could be to use a service mesh to provide that
+functionality.
+
+The `etcd` datastore of the control plane should have controls to limit access.
+Furthermore, mutual TLS (mTLS) should be used to communicate securely with it.
+The certificate authority for this should be unique to `etcd`.
+
```

Comment:
In addition, the Internet access to the Kubernetes API server should be restricted. 
If a Cloud provider is used for hosting Kubernetes, the access from pods to the Cloud metadata API (169.254.196.254) should also be restricted, or blocked if not needed.

Reply:
I could not find any documentation related to "not expose the API server publicly" in the docs, only in cloud providers' docs! Neither for the Cloud metadata API thing that is leaking stuff. It would definitely be a good addition! Thanks a lot!

Reply:
```suggestion
In addition, external Internet access to the Kubernetes API server should be restricted to not expose the API publicly. You can then use a bastion host to access the server. If a cloud provider is used for hosting Kubernetes, the access from pods to the cloud metadata API (169.254.196.254) should also be restricted or blocked if not needed because it can leak information.
```

Reply:
yes, it should be 169.254.169.254. Thanks a lot!

Reply:
```suggestion
In addition, external Internet access to the Kubernetes API server should be restricted
to not expose the API publicly. You can then use a bastion host to access the server. 
If a cloud provider is used for hosting Kubernetes, the access from pods to the cloud 
metadata API `169.254.169.254` should also be restricted or blocked if not needed 
because it may leak information.
```